### ISSUE TITLE
ver 1.4.4

### ISSUE THREAD

**Issue by region51** [barrust/Wikipedia#7]
```

Traceback (most recent call last):
  File "setup.py", line 6, in <module>
    from wikipedia import get_version
  File "/builddir/build/BUILD/Wikipedia-1.4.4/wikipedia/__init__.py", line 1, in <module>
    from .wikipedia import *
  File "/builddir/build/BUILD/Wikipedia-1.4.4/wikipedia/wikipedia.py", line 1187, in <module>
    _get_site_info()
  File "/builddir/build/BUILD/Wikipedia-1.4.4/wikipedia/wikipedia.py", line 1101, in _get_site_info
    'siprop': 'extensions|general'
  File "/builddir/build/BUILD/Wikipedia-1.4.4/wikipedia/wikipedia.py", line 1177, in _wiki_request
    r = WIKIPEDIA_GLOBALS['SESSION'].get(url, params=params, timeout=WIKIPEDIA_GLOBALS['TIMEOUT'])
  File "/usr/lib/python2.7/site-packages/requests/sessions.py", line 488, in get
    return self.request('GET', url, **kwargs)
  File "/usr/lib/python2.7/site-packages/requests/sessions.py", line 475, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/lib/python2.7/site-packages/requests/sessions.py", line 596, in send
    r = adapter.send(request, **kwargs)
  File "/usr/lib/python2.7/site-packages/requests/adapters.py", line 487, in send
    raise ConnectionError(e, request=request)
requests.exceptions.ConnectionError: HTTPConnectionPool(host='en.wikipedia.org', port=80): Max retries exceeded with url: /w/api.php?action=query&meta=siteinfo&siprop=extensions%7Cgeneral&format=json (Caused by NewConnectionError('<requests.packages.urllib3.connection.HTTPConnection object at 0xf6334a8c>: Failed to establish a new connection: [Errno -2] Name or service not known',))
RPM build errors:
error: Bad exit status from /var/tmp/rpm-tmp.cyFHLJ (%build)
    Bad exit status from /var/tmp/rpm-tmp.cyFHLJ (%build)
Child return code was: 1
EXCEPTION: Command failed. See logs for output.
 # bash --login -c /usr/bin/rpmbuild -bb --target noarch --nodeps  /builddir/build/SPECS/python-wikipedia.spec 
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/mockbuild/trace_decorator.py", line 88, in trace
    result = func(*args, **kw)
  File "/usr/lib/python2.7/site-packages/mockbuild/util.py", line 547, in do
    raise exception.Error("Command failed. See logs for output.\n # %s" % (command,), child.returncode)
Error: Command failed. See logs for output.
 # bash --login -c /usr/bin/rpmbuild -bb --target noarch --nodeps  /builddir/build/SPECS/python-wikipedia.spec 
LEAVE do --> EXCEPTION RAISED
```
:-1:

**Comment by barrust** [comment]
Are you not online? It looks like it failed to get basic wikipedia information. Installation worked for me using setup.py.

**Comment by barrust** [comment]
If it is that you are not online, I think there is a simple fix to get around that. I can do the site info pull as needed instead of on module import. I put this in a branch called [1.4.4-fix](https://github.com/barrust/Wikipedia/tree/1.4.4-fix) if you would like to try that. 

Honestly, I need to restructure the project into a class object which will break backward compatibility. That would allow for many of these issues to go away and would reduce some of the complexity of the code base.

**Comment by barrust** [comment]
PR #9 should resolve this issue.

**Comment by barrust** [comment]
This will be resolved in version 1.4.5
